packages/pretty-reporter/src/collapse-diagnostics.ts

```diff
@@ -0,0 +1,45 @@
+import { Diagnostic } from '@steiger/types'
+
+const DIAGNOSTIC_QUOTA = 20
+
+function distributeQuota(buckets: Array<number>, quota: number) {
+  const allItemsCount = buckets.reduce((acc, bucket) => acc + bucket, 0)
+  const quotaPerBucket = buckets.slice(0).fill(0)
+  let remainingQuota = Math.min(quota, allItemsCount)
+  let i = 0
+
+  while (remainingQuota > 0) {
+    const numOfItemsInBucket = buckets[i]
+    const assignedQuotaForBucket = quotaPerBucket[i]
+
+    // If it went beyond the last bucket, start from the first one
+    if (numOfItemsInBucket === undefined) {
+      i = 0
+      continue
+    }
+
+    // If the bucket already has the quota distributed for it or does not contain any items, skip it
+    if (assignedQuotaForBucket < numOfItemsInBucket && numOfItemsInBucket !== 0) {
+      quotaPerBucket[i] += 1
+      remainingQuota -= 1
+    }
```

Comment:
suggestion (non-blocking): I believe we can remove the `numOfItemsInBucket !== 0` part because `assignedQuotaForBucket` is always non-negative, so if `numOfItemsInBucket == 0`, then the first part of the condition will fail too (non-negative numbers are never less than zero).

I also suggest removing the comment for two reasons:
- with just one condition in the `if`, I think the code is pretty self-explanatory
- the comment describes the opposite of what's happening in the code below, which is a bit counter-intuitive
```suggestion
    if (assignedQuotaForBucket < numOfItemsInBucket) {
      quotaPerBucket[i] += 1
      remainingQuota -= 1
    }
```

Reply:
I agree, you're right